opengever/maintenance/scripts/sql_contact_migration.py

```diff
@@ -48,6 +57,49 @@ def export(self):
         org_roles = list(self.get_org_roles())
         self.export_json('memberships.json', org_roles)
 
+    def migrate_participations(self):
+        catalog = api.portal.get_tool('portal_catalog')
+        brains = catalog.unrestrictedSearchResults(
+            object_provides=IDossierMarker.__identifier__)
+        for brain in brains:
+            dossier = brain.getObject()
+            self.migrate_participation(dossier)
+
+    def migrate_participation(self, dossier):
+        sql_handler = SQLParticipationHandler(dossier)
+        kub_handler = KuBParticipationHandler(dossier)
+        participations = sql_handler.get_participations()
+
+        for participation in participations:
+            # To avoid reindexing the dosiser after each participation we
+            # add the participation manually
+            if isinstance(participation, OgdsUserParticipation):
+                participant_id = participation.ogds_userid
+
+            elif isinstance(participation, ContactParticipation):
+                if participation.contact.contact_type == 'person':
+                    participant_id = u'person:{}'.format(
+                        self.contact_mapping[participation.contact_id])
+                else:
+                    participant_id = u'organization:{}'.format(
+                        self.contact_mapping[participation.contact_id])
+
+            elif isinstance(participation, OrgRoleParticipation):
+                participant_id = u'membership:{}'.format(
+                    self.org_role_mapping[participation.org_role_id])
+            else:
+                raise Exception(
+                    u'Not supported participation type: {}'.format(
+                        participation.participation_type))
+
+            kub_participation = kub_handler.create_participation(
+                participant_id=participant_id,
+                roles=[role.role for role in participation.roles])
+            kub_handler.append_participation(kub_participation)
+
+        if participations:
+            dossier.reindexObject(idxs=["participations", "UID"])
```

Comment:
Ah you're right. I'll enable KUB before the participation migration.

Reply:
Ah yes, you're not using the `IParticipationAware` adapter, so that you are in control of which `ParticipationHandler` you are using in the rest of the script. So enabling KuB before should work.